lib/logmsg/tests/test_timestamp_serialize.c

```diff
@@ -22,78 +22,75 @@
  *
  */
 
-#include "testutils.h"
+#include "grab-logging.h"
+#include <criterion/criterion.h>
 #include "apphook.h"
 #include "logmsg/logmsg.h"
 #include "logmsg/timestamp-serialize.h"
 
-#define PREPARE_TEST UnixTime input_timestamps[LM_TS_MAX]; \
-  UnixTime output_timestamps[LM_TS_MAX]; \
-  GString *stream = g_string_new(""); \
-  input_timestamps[LM_TS_STAMP].ut_sec = 1; \
-  input_timestamps[LM_TS_STAMP].ut_usec = 2; \
-  input_timestamps[LM_TS_STAMP].ut_gmtoff = 3; \
-  input_timestamps[LM_TS_RECVD].ut_sec = 4; \
-  input_timestamps[LM_TS_RECVD].ut_usec = 5; \
-  input_timestamps[LM_TS_RECVD].ut_gmtoff = 6; \
-  input_timestamps[LM_TS_PROCESSED].ut_sec = 255; \
-  input_timestamps[LM_TS_PROCESSED].ut_usec = 255; \
-  input_timestamps[LM_TS_PROCESSED].ut_gmtoff = -1; \
-  SerializeArchive *sa = serialize_string_archive_new(stream);
+SerializeArchive *sa = NULL;
+GString *stream = NULL;
+UnixTime input_timestamps[LM_TS_MAX];
+UnixTime output_timestamps[LM_TS_MAX];
 
-#define CLEAN_TEST serialize_archive_free(sa); \
-  g_string_free(stream, TRUE);
-
-static void
-test_normal_working(void)
+Test(template_timestamp, test_normal_working)
 {
-  PREPARE_TEST
-  assert_true(timestamp_serialize(sa, input_timestamps), "Failed to serialize timestamps");
-  assert_true(timestamp_deserialize(sa, output_timestamps), "Failed to deserialize timestamps");
+  cr_assert(timestamp_serialize(sa, input_timestamps), "Failed to serialize timestamps");
+  cr_assert(timestamp_deserialize(sa, output_timestamps), "Failed to deserialize timestamps");
 
-  assert_nstring((const gchar *)input_timestamps, sizeof(input_timestamps),
-                 (const gchar *)output_timestamps, sizeof(output_timestamps),
-                 "The serialized and the deserialized timestamps are not equal");
-
-  CLEAN_TEST
+  cr_assert_str_eq((const gchar *)input_timestamps, (const gchar *)output_timestamps,
+                   "The serialized and the deserialized timestamps are not equal");
 }
 
-static void
-test_derializing_injured_timestamp(void)
+Test(template_timestamp, test_derializing_injured_timestamp)
 {
-  PREPARE_TEST
-
-  assert_true(timestamp_serialize(sa, input_timestamps), "Failed to serialize timestamps");
+  start_grabbing_messages();
```

Comment:
Do we need to use grab message functionality here ?

Reply:
@Kokan : I can see this functionality in the previous version of code.

Reply:
Yes, you are correct.